dora_node_hub/dora_diff_drive_controller/src/odometry.rs

```diff
@@ -0,0 +1,138 @@
+use crate::pose_2d::Pose2D;
+
+fn normalize_angle(angle: f64) -> f64 {
+    let result = (angle + std::f64::consts::PI) % (2.0 * std::f64::consts::PI);
+    if result <= 0.0 {
+        result + std::f64::consts::PI
+    } else {
+        result - std::f64::consts::PI
+    }
+}
+/// Odometry for a diff drive mobile robot.
+pub struct DiffDriveOdometry {
+    /// Current pose:
+    pub current_pose: Pose2D,
+
+    /// Current velocity:
+    pub linear: f64, // [m/s]
+    pub angular: f64, // [rad/s]
+
+    /// The distance between the wheels.
+    wheel_separation: f64, // [m]
+    /// The radius of the wheels.
+    wheel_radius: f64, // [m]
+
+    /// Previous data for odometry calculations.
+    previous_time: f64, // [s]
+    previous_left_wheel_position: f64,  // [rad]
+    previous_right_wheel_position: f64, // [rad]
+}
+
+impl DiffDriveOdometry {
+    pub fn new(wheel_separation: f64, wheel_radius: f64) -> Self {
```

Comment:
Nit: add some docstrings
```suggestion
    /// Creates an instance of `DiffDriveOdometry`.
    ///
    /// # Arguments
    /// * `wheel_separation` - The distance between the wheels in meters.
    /// * `wheel_radius` - The radius of the wheels in meters.
    pub fn new(wheel_separation: f64, wheel_radius: f64) -> Self {
```

Reply:
Done!